x/cellarfees/keeper/pool.go

```diff
@@ -0,0 +1,56 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/peggyjv/sommelier/v4/app/params"
+	"github.com/peggyjv/sommelier/v4/x/cellarfees/types"
+)
+
+// Appends the coin to the pool coins if denom isn't already present, otherwise add the amount to the
+// existing balance
+func (k Keeper) AddCoinToPool(ctx sdk.Context, coin sdk.Coin) {
+	// (Collin) should an empty coin error?
+	if (coin == sdk.Coin{} || coin.IsZero()) {
+		return
+	}
+
+	if coin.Denom == params.BaseCoinUnit {
+		panic("Cannot add SOMM to cellar fee pool")
+	}
+
+	pool := k.GetCellarFeePool(ctx)
+	pool.Pool = pool.Pool.Add(coin)
+	k.SetCellarFeePool(ctx, pool)
+}
+
+// Appends the coin to the pool coins if denom isn't already present, otherwise add the amount to the
+// existing balance
+func (k Keeper) AddCoinsToPool(ctx sdk.Context, coins sdk.Coins) {
+	// (Collin) should this error?
+	if len(coins) == 0 {
+		return
+	}
+
+	// don't add usomm to the pool
+	if !coins.AmountOfNoDenomValidation(params.BaseCoinUnit).IsZero() {
+		panic("Cannot add SOMM to cellar fee pool")
```

Comment:
Similar comment around potentially erroring instead

Reply:
I realize now this method isn't actually used. I'll mark it for removal in case review elsewhere results in changes that require it.